packages/manager/core/generator/app/templates/src/components/layout-helpers/Listing/DatagridWrapper.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { OsdsDatagrid } from '@ovhcloud/ods-component-datagrid/react';
+import { OsdsLink } from '@ovhcloud/ods-component-link/react';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import appConfig from '@/{{appName}}.config';
+import ConfigInterface from '@/configInterface';
+import { ReactFormatter } from '@ovh-ux/manager-core-utils';
+
+function LinkService({ cellData, navigate }: any) {
+  return (
+    <OsdsLink
+      color={ODS_THEME_COLOR_INTENT.primary}
+      onClick={() => navigate(`/${cellData}`)}
+    >
+      {cellData}
+    </OsdsLink>
+  );
+}
+
+export default function DatagridWrapper({ data }: any) {
+  const navigate = useNavigate();
+  const myConfig: ConfigInterface = appConfig;
+  const serviceKey = myConfig.listing?.datagrid?.serviceKey;
+  const tmp = Object.assign(data, {});
+  if (tmp && tmp.length === 0) return <div>null</div>;
+  const keys = Object.keys(tmp[0]);
+  const columns = keys
+    .filter((element: any) => {
+      if (element === 'iam') return false;
+      return true;
+    })
+    .map((elem: any) => ({
+      title: elem,
+      field: elem,
+      isSortable: false,
+      formatter:
+        serviceKey && elem === serviceKey
+          ? ReactFormatter(<LinkService navigate={navigate} />)
```

Comment:
```suggestion
          ? ReactFormatter(<LinkService onClick={() => navigate(`/${cellData}`)} />
)
```

Reply:
you can get the cell data only inside the component LinkService

Reply:
```suggestion
          ? ReactFormatter(
              <LinkService
                onClick={({ cellData }: any) => {
                  navigate(`/${cellData}`);
                }}
              />,
            )
```